c7n/resources/ecs.py

```
@@ -326,8 +326,88 @@ def process_cluster_resources(self, client, cluster_id, container_instances):
+        # update_container_instances_state call is 10
+        chunk_size = 10
+        client = local_session(self.manager.session_factory).client('ecs')
+        for service_set in chunks(c_instances, chunk_size):
+            try:
+                client.update_container_instances_state(
+                    cluster = cluster,
+                    containerInstances = service_set,
+                    status = self.data.get('state'))
+            except ClientError:
+                self.manager.log.warning(
+                    'Failed to update Container Instances State: %s, cluster %s' %
+                    (service_set, cluster))
+                raise
+
+
+@ContainerInstance.action_registry.register('update-agent')
+class UpdateAgent(BaseAction):
+    """Updates the agent on a container instance
+    """
+
+    schema = type_schema('update-agent')
+    permissions = ('ecs:UpdateContainerAgent',)
+
+    def process(self, resources):
+        for r in resources:
+            results = self.process_instance(
+                r.get('c7n:cluster'),
+                r.get('containerInstanceArn'))
+            return results
+
+    def process_instance(self, cluster, instance):
+        client = local_session(self.manager.session_factory).client('ecs')
+        try:
+            client.update_container_agent(
+                cluster = cluster,
+                containerInstance = instance)
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'NoUpdateAvailableException':
+                self.manager.log.warning(
+                    'No update available for Container Instance: %s, cluster: %s'
+                    % (instance, cluster))
+            if e.response['Error']['Code'] == 'UpdateInProgressException':
+                self.manager.log.warning(
+                    'Container Instance Agent update already in progress: %s, cluster %s' %
+                    (instance, cluster))
```

this is still silently swallowing unknown exceptions, it should raise on unknown.